drivers/storage/portworx/portworx.go

```diff
@@ -153,6 +155,15 @@ func (p *portworx) GetSelectorLabels() map[string]string {
 }
 
 func (p *portworx) SetDefaultsOnStorageCluster(toUpdate *corev1.StorageCluster) {
+	if cloudprovider.IsEKS() {
+		toUpdate.Annotations[pxutil.AnnotationIsEKS] = "true"
+		// Do preflight check for eks cloud permission once before px installation
+		// TODO: use stc status instead of annotation? but annotation can be set by UI in the future
```

Comment:
I feel we may still need the annotation, because we need a switch to control the preflight, to the user, to future UI. If not using annotation we may need another spec for the same purpose.